utilities/object_registry.cc

```diff
@@ -135,14 +136,19 @@ void ObjectLibrary::Dump(Logger *logger) const {
 // Returns the Default singleton instance of the ObjectLibrary
 // This instance will contain most of the "standard" registered objects
 std::shared_ptr<ObjectLibrary> &ObjectLibrary::Default() {
-  static std::shared_ptr<ObjectLibrary> instance =
-      std::make_shared<ObjectLibrary>("default");
+  // Use avoid destruction here so the default ObjectLibrary will not be
+  // statically destroyed and long-lived.
+  STATIC_AVOID_DESTRUCTION(std::shared_ptr<ObjectLibrary>, instance)
+  (std::make_shared<ObjectLibrary>("default"));
   return instance;
 }
 
 std::shared_ptr<ObjectRegistry> ObjectRegistry::Default() {
-  static std::shared_ptr<ObjectRegistry> instance(
-      new ObjectRegistry(ObjectLibrary::Default()));
+  // Use avoid destruction here so the default ObjectRegistry will not be
+  // statically destroyed and long-lived.
+  STATIC_AVOID_DESTRUCTION(ObjectRegistry, registry)(ObjectLibrary::Default());
+  STATIC_AVOID_DESTRUCTION(std::shared_ptr<ObjectRegistry>, instance)
+  (&registry);
```

Comment:
As you can see in CircleCI, this does not pass ASAN, because it's not a valid way to construct a destructible shared_ptr.